infrastructure/main.tf

```diff
@@ -1,136 +1,139 @@
-    REFORM_SERVICE_NAME = "${local.app_full_name}"
-    REFORM_TEAM = "${var.team_name}"
-    REFORM_SERVICE_TYPE = "${var.app_language}"
-    REFORM_ENVIRONMENT = "${var.env}"
+    REFORM_SERVICE_NAME = local.app_full_name
+    REFORM_TEAM = var.team_name
+    REFORM_SERVICE_TYPE = var.app_language
+    REFORM_ENVIRONMENT = var.env
 
-    PACKAGES_NAME = "${local.app_full_name}"
-    PACKAGES_PROJECT = "${var.team_name}"
-    PACKAGES_ENVIRONMENT = "${var.env}"
+    PACKAGES_NAME = local.app_full_name
+    PACKAGES_PROJECT = var.team_name
+    PACKAGES_ENVIRONMENT = var.env
 
-    LOG_OUTPUT = "${var.log_output}"
+    LOG_OUTPUT = var.log_output
   }
 }
 
 module "db" {
   source = "git@github.com:hmcts/cnp-module-postgres?ref=master"
   product = "${local.app_full_name}-postgres-db"
-  location = "${var.location}"
-  env = "${var.env}"
-  postgresql_user = "${var.postgresql_user}"
-  database_name = "${var.database_name}"
+  location = var.location
+  env = var.env
+  postgresql_user = var.postgresql_user
+  database_name = var.database_name
   sku_name = "GP_Gen5_2"
   sku_tier = "GeneralPurpose"
   storage_mb = "51200"
-  common_tags  = "${var.common_tags}"
+  common_tags  = var.common_tags
 }
 
 data "azurerm_key_vault" "shared_key_vault" {
-  name = "${local.shared_vault_name}"
-  resource_group_name = "${local.shared_vault_name}"
+  name = local.shared_vault_name
+  resource_group_name = local.shared_vault_name
+}
+
+data "azurerm_key_vault" "s2s_vault" {
+  name = "s2s-${local.local_env}"
+  resource_group_name = "rpe-service-auth-provider-${local.local_env}"
 }
 
 data "azurerm_key_vault_secret" "s2s_secret" {
   name = "microservicekey-coh-cor"
-  vault_uri = "${local.s2s_vault_url}"
+  key_vault_id = data.azurerm_key_vault.s2s_vault.id
 }
 
-//data "azurerm_key_vault_secret" "oauth2_secret" {
-//  name = "cor-oauth2-token"
-//  vault_uri = "${data.azurerm_key_vault.shared_key_vault.vault_uri}"
-//}
-
 module "local_key_vault" {
   source = "git@github.com:hmcts/cnp-module-key-vault?ref=master"
-  product = "${local.app_full_name}"
-  env = "${var.env}"
-  tenant_id = "${var.tenant_id}"
-  object_id = "${var.jenkins_AAD_objectId}"
-  resource_group_name = "${module.app.resource_group_name}"
+  product = local.app_full_name
+  env = var.env
+  tenant_id = var.tenant_id
+  object_id = var.jenkins_AAD_objectId
+  resource_group_name = module.app.resource_group_name
   product_group_object_id = "5d9cd025-a293-4b97-a0e5-6f43efce02c0"
-  common_tags  = "${var.common_tags}"
+  common_tags  = var.common_tags
+}
+
+data "azurerm_key_vault" "local_key_vault" {
+  name = local.vaultName
+  resource_group_name = local.vaultName
 }
 
 resource "azurerm_key_vault_secret" "POSTGRES-USER" {
   name = "${local.app_full_name}-POSTGRES-USER"
-  value = "${module.db.user_name}"
-  vault_uri = "${module.local_key_vault.key_vault_uri}"
+  value = module.db.user_name
+  key_vault_id = data.azurerm_key_vault.local_key_vault.id
```

Comment:
I know this is the new syntax, but for myself, jenkins failed as it couldnt read key_vault_id

Reply:
we'll try it out